internal/cmd/controller/clustermonitor/monitor_test.go

```diff
@@ -0,0 +1,369 @@
+package clustermonitor_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/rancher/fleet/internal/cmd/controller/clustermonitor"
+	"github.com/rancher/fleet/internal/mocks"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/wrangler/v3/pkg/genericcondition"
+)
+
+const threshold = 1 * time.Second
+
+// BDStatusMatcher implements a gomock matcher for bundle deployment status.
+// The matcher checks for empty modified and non-ready status, as well as offline state through Ready and Monitored
+// conditions.
+type BDStatusMatcher struct {
+}
+
+func (m BDStatusMatcher) Matches(x interface{}) bool {
+	bd, ok := x.(*v1alpha1.BundleDeployment)
+	if !ok || bd == nil {
+		return false
+	}
+
+	bdStatus := bd.Status
+
+	if bdStatus.ModifiedStatus != nil || bdStatus.NonReadyStatus != nil {
+		return false
+	}
+
+	foundReady, foundMonitored := false, false
+	for _, cond := range bdStatus.Conditions {
+		if cond.Type == "Ready" {
+			foundReady = true
+
+			if !strings.Contains(cond.Message, "offline") {
+				return false
+			}
+		} else if cond.Type == "Monitored" {
+			foundMonitored = true
+
+			if !strings.Contains(cond.Message, "offline") {
+				return false
+			}
```

Comment:
Would be good to check that the condition's status is `False` when the `offline` string is found and `True` otherwise. 
Also would be good to check that the Reason in the Condition.

Reply:
Thanks, checks added.